This patch fixes an issue that can arise when using the AES opcodes
on a Rabbit 6000.  AES opcodes on Rabbit 6000 require canonical
ordering of Px/Px' registers.  We maintain the ordering as well as
possible, but need to ensure that task switching doesn't throw it off.

The Dynamic C 10.72F release includes the patched UCOS2.LIB file, but
the uC/OS-II source isn't included in the Open Source GitHub code
repository.  If your Dynamic C installation is based on 10.72E or
earlier, you will need to manually apply this fix using `patch`:

    patch -p1 --forward < ucos2-patch2.patch

Include the command-line option "--dry-run" to test the patch.  The
"--forward" option ignores the patch if already applied.

--- a/Lib/Rabbit4000/UCOS2/UCOS2.LIB
+++ b/Lib/Rabbit4000/UCOS2/UCOS2.LIB
@@ -2652,6 +2652,15 @@ OSCtxSw::
    ex    af, af'
    push  af
    exx
+#if _RAB6K
+   push  jkhl
+   push  bcde
+   push  iy
+   alts push  pw
+   alts push  px
+   alts push  py
+   alts push  pz
+#else
    exp
    push  jkhl
    push  bcde
@@ -2660,13 +2669,21 @@ OSCtxSw::
    push  px
    push  py
    push  pz
-
+#endif
 	; Suspend this task and start running the highest-priority task.  We will
 	; resume the suspended task when this function or the periodic ISR swaps
 	; our stack back and performs an LRET.
    lcall oscs_1
 
-	; Resuming current task, need to restore registers from stack.
+#if _RAB6K
+   altd pop   pz
+   altd pop   py
+   altd pop   px
+   altd pop   pw
+   pop   iy
+   pop   bcde
+   pop   jkhl
+#else
    pop   pz
    pop   py
    pop   px
@@ -2675,6 +2692,8 @@ OSCtxSw::
    pop   bcde
    pop   jkhl
    exp
+#endif
+	; Resuming current task, need to restore registers from stack.
    exx
    pop   af
    ex    af, af'

